stan/math/opencl/kernels/tridiagonalization.hpp

```diff
@@ -0,0 +1,326 @@
+#ifndef STAN_MATH_GPU_KERNELS_TRIDIAGONALIZATION_HPP
+#define STAN_MATH_GPU_KERNELS_TRIDIAGONALIZATION_HPP
+
+#ifdef STAN_OPENCL
+
+#include <stan/math/opencl/kernel_cl.hpp>
+
+namespace stan {
+namespace math {
+namespace opencl_kernels {
+
+// \cond
+const char* tridiagonalization_householder_kernel_code = STRINGIFY(
+    // \endcond
+    /**
+     * Calculates householder vector and first element of the vector v.
+     * Must be run with 1 workgroup of 1024 threads.
```

Comment:
What happens with devices that dont support 1024 threads in a thread block?

Reply:
Nothing good. Do we need to support any devices that have lower maximum number of threads in a work group? As far as I know all recent CPUs and GPUs can run 1024.

Reply:
Yeah, I only spotted < 1024 on my ancient integrated Intel GPU. Was just wondering. We should not really care about conforming with GPUs where this wont be any faster anyways.

Reply:
Oh, I forgot about integrated GPUs. I think making it possible to run smaller number of threads per group wont be too much work. What is the maximum for your iGPU? I cant find any official numbers for intel GPUs.

Reply:
Its 256 typically, thats also what older AMD GPUs supported.

Reply:
I checked the code for this. Actually it will run just fine with smaller local size.